python/ray/tune/trial_runner.py

```diff
@@ -674,9 +676,18 @@ def stop_trial(self, trial):
         if trial.status in [Trial.ERROR, Trial.TERMINATED]:
             return
         elif trial.status in [Trial.PENDING, Trial.PAUSED]:
-            self._scheduler_alg.on_trial_remove(self, trial)
-            self._search_alg.on_trial_complete(
-                trial.trial_id, early_terminated=True)
+            try:
+                result = self.trial_executor.fetch_result(trial)
+                trial.update_last_result(result, terminate=True)
+                self._scheduler_alg.on_trial_remove(self, trial)
+                self._search_alg.on_trial_complete(
+                    trial.trial_id, result=result, early_terminated=True)
```

Comment:
I think we should probably leave this as is; calling `stop_trial` should probably not block.